hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestCOWDataSource.scala

```diff
@@ -1860,14 +1860,16 @@ class TestCOWDataSource extends HoodieSparkClientTestBase with ScalaAssertionSup
         }
         if (firstClusteringState == HoodieInstant.State.REQUESTED.name()) {
           val table = HoodieSparkTable.create(writeConfig, context)
-          table.rollbackInflightClustering(
-            metaClient.getActiveTimeline.getLastClusteringInstant.get,
-            new java.util.function.Function[String, Option[HoodiePendingRollbackInfo]] {
-              override def apply(commitToRollback: String): Option[HoodiePendingRollbackInfo] = {
-                new SparkRDDWriteClient(context, writeConfig).getTableServiceClient
-                  .getPendingRollbackInfo(table.getMetaClient, commitToRollback, false)
-              }
-            })
+          val client = new SparkRDDWriteClient(context, writeConfig)
```

Comment:
the client is created just to fetch the txn manager?

Reply:
No. It is originally created in line 1867. It is used for the call to `getPendingRollbackInfo`